tempodb/encoding/vparquet3/schema.go

```diff
@@ -49,18 +49,33 @@ const (
 	DefinitionLevelResourceAttrs             = 2
 	DefinitionLevelResourceSpansILSSpan      = 3
 	DefinitionLevelResourceSpansILSSpanAttrs = 4
+	// TODO do we need a new definition level for attr values?
 
 	FieldResourceAttrKey       = "rs.list.element.Resource.Attrs.list.element.Key"
-	FieldResourceAttrVal       = "rs.list.element.Resource.Attrs.list.element.Value"
-	FieldResourceAttrValInt    = "rs.list.element.Resource.Attrs.list.element.ValueInt"
-	FieldResourceAttrValDouble = "rs.list.element.Resource.Attrs.list.element.ValueDouble"
-	FieldResourceAttrValBool   = "rs.list.element.Resource.Attrs.list.element.ValueBool"
+	FieldResourceAttrVal       = "rs.list.element.Resource.Attrs.list.element.Value.list.element"
+	FieldResourceAttrValInt    = "rs.list.element.Resource.Attrs.list.element.ValueInt.list.element"
+	FieldResourceAttrValDouble = "rs.list.element.Resource.Attrs.list.element.ValueDouble.list.element"
+	FieldResourceAttrValBool   = "rs.list.element.Resource.Attrs.list.element.ValueBool.list.element"
 
 	FieldSpanAttrKey       = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.Key"
-	FieldSpanAttrVal       = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.Value"
-	FieldSpanAttrValInt    = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueInt"
-	FieldSpanAttrValDouble = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueDouble"
-	FieldSpanAttrValBool   = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueBool"
+	FieldSpanAttrVal       = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.Value.list.element"
+	FieldSpanAttrValInt    = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueInt.list.element"
+	FieldSpanAttrValDouble = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueDouble.list.element"
+	FieldSpanAttrValBool   = "rs.list.element.ss.list.element.Spans.list.element.Attrs.list.element.ValueBool.list.element"
+)
+
+type AttrType int32
```

Comment:
byte? or uint16?

Reply:
The smallest integer size in parquet go is int32 / uint32

Reply:
You are right. You can do a `[]byte` but not just a `byte` which is odd. I think ideally there's a few fields we'd encode in smaller sizes now that i'm going back through the schema.

This is fine.